crates/cubecl-linalg/src/convolution/reader/im2col.rs

```diff
@@ -109,31 +112,63 @@ impl<E: Numeric> Im2colReader<E> {
         let view_m = view_tile_m + load_m;
         let view_k = view_tile_k + load_k;
 
-        let (out_nh, out_x) = self.shape_out_x.div_mod(view_m);
-        let (batch, out_y) = self.shape_out_y.div_mod(out_nh);
-
-        let kernel_w = config.kernel_size(1);
+        let (batch, out_offs) = div_mod_seq(view_m, &self.shape_out);
 
         let channel = view_k % self.shape_channel;
-        let rem = view_k / self.shape_channel;
-        let kernel_x = rem % kernel_w;
-        let kernel_y = rem / kernel_w;
+        let mut rem = view_k / self.shape_channel;
+
+        let spatial_dims = comptime![self.shapes_spatial.len()];
+        let mut in_pos = Sequence::<i32>::new();
+
+        #[unroll]
+        for i in 0..spatial_dims {
```

Comment:
I would document a bit more what is the spatial dims.

Reply:
I think this might just be a language barrier. It's spatial, as in space, as in depth, height, width or z, y, x. As opposed to the batch and channels, which don't represent a position in space. It should be pretty self-explanatory.

Reply:
So anything not batch related

Reply:
Not batch and not channels, yes.